lib/hex_web/user.ex

```diff
@@ -90,6 +93,38 @@ defmodule HexWeb.User do
     |> HexWeb.Repo.update
   end
 
+  def initiate_password_reset(user) do
+    key = gen_key()
+    now = Util.ecto_now
+
+    %{user | reset_key: key, reset_expiry: now, updated_at: now}
+    |> HexWeb.Repo.update
+
+    send_reset_email(user, key)
+  end
+
+  def reset?(username, key, password) do
+    if (user = get(username: username)) && user.reset_key == key && Util.within_last_day(user.reset_expiry) do
```

Comment:
We should be doing this in `confirm?` as well. Eric has a secure compare function in Util, so I'll use that.